src/main/java/de/hysky/skyblocker/skyblock/fancybars/StatusBar.java

```diff
@@ -0,0 +1,353 @@
+package de.hysky.skyblocker.skyblock.fancybars;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import de.hysky.skyblocker.SkyblockerMod;
+import de.hysky.skyblocker.utils.render.RenderHelper;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.*;
+import net.minecraft.client.gui.screen.narration.NarrationMessageBuilder;
+import net.minecraft.client.gui.widget.ClickableWidget;
+import net.minecraft.client.gui.widget.Widget;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.function.Consumer;
+
+public class StatusBar implements Widget, Drawable, Element, Selectable {
+
+    private static final Identifier BAR_FILL = new Identifier(SkyblockerMod.NAMESPACE, "bars/bar_fill");
+    private static final Identifier BAR_BACK = new Identifier(SkyblockerMod.NAMESPACE, "bars/bar_back");
+
+
+   /* public static final Codec<StatusBar> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+                    Codec.INT.fieldOf("size").forGetter(bar -> bar.size),
+                    Codec.INT.fieldOf("x").forGetter(bar -> bar.gridX),
+                    Codec.INT.fieldOf("y").forGetter(bar -> bar.gridY),
+                    Codec.STRING.listOf().fieldOf("colors").xmap(
+                                    strings -> strings.stream().map(s -> Integer.parseInt(s, 16)).map(Color::new).toArray(Color[]::new),
+                                    colors -> Arrays.stream(colors).map(color -> Integer.toHexString(color.getRGB())).toList())
+                            .forGetter(StatusBar::getColors),
+                    Codec.STRING.optionalFieldOf("text_color").xmap(
+                                    s -> {
+                                        if (s.isPresent()) {
+                                            return Optional.of(new Color(Integer.parseInt(s.get(), 16)));
+                                        } else return Optional.empty();
+                                    },
+                                    o -> o.map(object -> Integer.toHexString(((Color) object).getRGB())))
+                            .forGetter(bar -> {
+                                if (bar.getTextColor() != null) {
+                                    return Optional.of(bar.getTextColor());
+                                } else return Optional.empty();
+                            }),
+                    Codec.BOOL.optionalFieldOf("show_text", true).forGetter(StatusBar::showText),
+                    Codec.STRING.fieldOf("icon_position").xmap(
+                            IconPosition::valueOf,
+                            Enum::toString
+                    ).forGetter(bar -> bar.iconPosition)
+            )
+
+            .apply(instance, ));*/
```

Comment:
Uh what's this?

Reply:
sad attempt at a codec for the config. Left it in just in case one of you guys wanted to make something out of it. So can I just remove it?